packages/project/lib/build/definitions/component.js

```diff
@@ -0,0 +1,128 @@
+import {enhancePatternWithExcludes} from "./_utils.js";
+import {enhanceBundlesWithDefaults} from "../../validation/validator.js";
+
+/**
+ * Get tasks and their configuration for a given component project
+ *
+ * @private
+ * @param {object} parameters
+ * @param {object} parameters.project
+ * @param {object} parameters.taskUtil
+ * @param {Function} parameters.getTask
+ */
+export default function({project, taskUtil, getTask}) {
+	const tasks = new Map();
+	tasks.set("escapeNonAsciiCharacters", {
+		options: {
+			encoding: project.getPropertiesFileSourceEncoding(),
+			pattern: "/**/*.properties"
+		}
+	});
+
+	tasks.set("replaceCopyright", {
+		options: {
+			copyright: project.getCopyright(),
+			pattern: "/**/*.{js,json}"
+		}
+	});
+
+	tasks.set("replaceVersion", {
+		options: {
+			version: project.getVersion(),
+			pattern: "/**/*.{js,json}"
+		}
+	});
+
+	// Support rules should not be minified to have readable code in the Support Assistant
```

Comment:
Support rules are AFAIK a concept for libraries, so I wonder why we also do this for applications. But as we do, I think we should also do it here to stay consistent.